languageApp/src/components/CardList/CardList.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import {FlatList} from 'react-native';
+//components
+import {ToReadCard} from '../ToReadCard/ToReadCard';
+import {ListeningCard} from '../ListeningCard/ListeningCard';
+//navigation
+import {useRoute} from '@react-navigation/native';
+//interfaces
+import {PodcastEpisode} from '../../interfaces/CardsInterfaces';
+import {Track} from '../../interfaces/CardsInterfaces';
+import {Story} from '../../interfaces/CardsInterfaces';
+
+interface CardListProps {
+  musicData?: Track[];
+  podcastData?: PodcastEpisode[];
+  storyData?: Story[];
+}
+
+const ToReadList: React.FC<{data: Story[]}> = ({data}) => (
+  <FlatList
+    data={data}
+    renderItem={({item}) => <ToReadCard {...item} />}
+    keyExtractor={(item, index) => index.toString()}
+  />
+);
+
+const PodcastList: React.FC<{data: PodcastEpisode[]}> = ({data}) => (
+  <FlatList
+    data={data}
+    renderItem={({item}) => <ListeningCard {...item} />}
+    keyExtractor={(item, index) => index.toString()}
+  />
+);
+
+const MusicList: React.FC<{data: Track[]}> = ({data}) => (
+  <FlatList
+    data={data}
+    renderItem={({item}) => <ListeningCard {...item} />}
+    keyExtractor={(item, index) => index.toString()}
+  />
+);
+
+const CardList: React.FC<CardListProps> = ({
+  musicData,
+  podcastData,
+  storyData,
+}) => {
+  const route = useRoute();
+
+  if (route.name === 'ToReadScreen') {
+    return <ToReadList data={storyData || []} />;
+  } else if (route.name === 'PodcastScreen') {
+    return <PodcastList data={podcastData || []} />;
+  } else {
+    return <MusicList data={musicData || []} />;
+  }
+};
```

Comment:
The if-else if-else block is a nice solution, but be aware that if options get increased, readability could suffer, and performance might be impacted due to checking each condition, if you expect more options and the logic isn't complex, consider using a switch or refactoring with an object {route.name: component} with return object[route.name] to improve readability